src/app/components/character-count/index.html

```diff
@@ -0,0 +1,284 @@
+
+          <sky-docs-visual-example-image>
+            <div>
+              <img
+                src="~/assets/img/guidelines/character-count/character-count-usage-1.png"
+              />
+              <div
+                class="caption caption-success sky-text-success"
+              >
+                Do use character count on input fields with character limits that a user could realistically exceed.
+              </div>
+            </div>
+          </sky-docs-visual-example-image>
+        </sky-docs-visual-example>
+
+        <h3
+          class="sky-text-danger"
+        >
+          Don't use when
+        </h3>
+
+        <sky-docs-visual-example>
+          <sky-docs-visual-example-text>
+            <p>
+              Don’t use character count on inputs where the character limit is apparent or implied by the data type.
+            </p>
+          </sky-docs-visual-example-text>
+
+          <sky-docs-visual-example-image>
+            <div>
+              <img
+                src="~/assets/img/guidelines/character-count/character-count-usage-2.png"
+              />
+              <div
+                class="caption caption-danger sky-text-danger"
+              >
+                Don't use character count when the content has an expected or standard length.
+              </div>
+            </div>
+          </sky-docs-visual-example-image>
+        </sky-docs-visual-example>
+
+        <sky-docs-visual-example>
+          <sky-docs-visual-example-text>
+            <p>
+              Don’t use character count on inputs when exceeding the character limit is unrealistic. Use normal form validation to handle errors caused by character limit violations.
+            </p>
+          </sky-docs-visual-example-text>
+
+          <sky-docs-visual-example-image>
+            <div
+              class="sky-margin-stacked-content-section"
+            >
+              <img
+                src="~/assets/img/guidelines/character-count/character-count-usage-3.png"
+              />
+              <div
+                class="caption caption-danger sky-text-danger"
+              >
+                Don't use character count when the character limit is unlikely to be exceeded.
+              </div>
+            </div>
+
+            <div>
+              <img
+                src="~/assets/img/guidelines/character-count/character-count-usage-4.png"
+              />
+              <div
+                class="caption caption-success sky-text-success"
+              >
+                Do use normal form validation for edge cases where a high character limit is exceeded.
+              </div>
+            </div>
+          </sky-docs-visual-example-image>
+        </sky-docs-visual-example>
+      </sky-demo-page-content>
+
+      <sky-demo-page-content sectionHeading="Anatomy">
+
+        <sky-docs-anatomy
+          imagePath="~/assets/img/guidelines/character-count/character-count-anatomy-1.png"
```

Comment:
#1 in this image is in the wrong place. It is more associated with the label than with the input. Looking at the code, I don't think the input is actually part of the component, so I wonder if it should be in this image at all. But if we want a label to show the standard presentation, I think we should still move the #1 to the right so that it is not so closely associated with the label and move it down to overlap with the top border of the input.

#4 is also oddly placed. Why is it overlapping the border? Is this because it is for the red border as well as the status indicator and icon? If so, then I still think it should be moved down a bit to overlap less of the input. But if it is only for the status indicator and message, then I think it should be moved down more.

Reply:
Redesigned the anatomy image.